inc/c_util/tarray.h

```diff
@@ -10,167 +10,38 @@
 #include <inttypes.h>
 #endif
 
-#include "thandle.h"
+#include "c_util/thandle_ll.h"
+#include "c_util/tarray_ll.h"
 
 #include "umock_c/umock_c_prod.h"
 
-/*TARRAY is backed by a THANDLE build on the structure below*/
-#define TARRAY_STRUCT_TYPE_NAME_TAG(T) MU_C2(TARRAY_TYPEDEF_NAME(T), _TAG)
-#define TARRAY_TYPEDEF_NAME(T) MU_C2(TARRAY_STRUCT_, T)
+#define TARRAY_CREATE_DECLARE(T) TARRAY_LL_CREATE_DECLARE(T, T)
+#define TARRAY_CREATE_DEFINE(T) TARRAY_LL_CREATE_DEFINE(T, T)
 
-/*TARRAY_TYPE(T) introduces the base type that holds an array of T*/
-#define TARRAY_DEFINE_STRUCT_TYPE(T)                                                \
-typedef struct TARRAY_STRUCT_TYPE_NAME_TAG(T)                                       \
-{                                                                                   \
-    uint32_t capacity;                                                              \
-    T* arr;                                                                         \
-} TARRAY_TYPEDEF_NAME(T);                                                           \
+#define TARRAY_ENSURE_CAPACITY_DECLARE(T) TARRAY_LL_ENSURE_CAPACITY_DECLARE(T, T)
+#define TARRAY_ENSURE_CAPACITY_DEFINE(T) TARRAY_LL_ENSURE_CAPACITY_DEFINE(T, T)
 
-/*TARRAY is-a THANDLE*/
-/*given a type "T" TARRAY(T) expands to the name of the type. */
-#define TARRAY(T) THANDLE(TARRAY_TYPEDEF_NAME(T))
+#define TARRAY_FREE_DEFINE(T) TARRAY_LL_FREE_DEFINE(T, T)
+
+#define TARRAY_CREATE(C) TARRAY_LL_CREATE(C)
+#define TARRAY_ENSURE_CAPACITY(C) TARRAY_LL_ENSURE_CAPACITY(C)
```

Comment:
After reading thandle.h, I'm wondering if most of these can be removed and the LL can just be directly used in lines 37-47. Do these need to be exposed with these names?

Reply:
TARRAY is just a wrapper over TARRAY_LL where C==T. I don't see a problem with they way these are written, as they are a bit of separation of concerns:
concern1 : how is TARRAY linked to TARRAY_LL (addressed here)

concern 2: what's the content of a TARRAY_DECLARE/DEFINE (addressed in lines 36-47)